torch/_ops.py

```diff
@@ -782,6 +775,101 @@ def tags(self):
     # TODO: add more methods to expose information about input and output arguments
 
 
+# TorchBindOpOverload are those custom ops which have at least one overload's
+# schema consists of torch.ScriptObject (i.e. custom class) input.
+# TorchBindOpOverload will skip C++ dispatcher and purely dispatched in python
+# when its inputs contain FakeScriptObject in a similar way as higher order ops.
+class TorchBindOpOverload(OpOverload):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def _fallthrough_keys(self) -> List[DispatchKey]:
+        # TODO: we should be calling the fallback for these, but a fallthrough is almost close
+        # enough to the fallback in most cases that we care about.
+        _DEFAULT_FALLTHROUGH_KEYS = [
+            DispatchKey.Autograd,
+            DispatchKey.AutogradCPU,
+            DispatchKey.AutogradCUDA,
+            DispatchKey.ADInplaceOrView,
+            DispatchKey.PythonTLSSnapshot,
+        ]
+
+        def _may_use_fallthrough_instead_of_fallback(key: DispatchKey):
+            _fallthrough_in_cpp_dispatcher = (
+                torch._C._dispatch_has_kernel_for_dispatch_key(self.name(), key)
+                and torch._C._dispatch_kernel_for_dispatch_key_is_fallthrough(
+                    self.name(), key
+                )
+            )
+
+            _fallthrough_in_python_dispatcher = (
+                key not in self.py_kernels
+                or self.py_kernels[key] is torch.library.fallthrough_kernel
+            )
+            return _fallthrough_in_cpp_dispatcher or _fallthrough_in_python_dispatcher
```

Comment:
This is not correct -- if the key is not in self.py_kernels then this returns True, even if there's a C++ kernel for the DispatchKey. We should get rid of that condition.

Reply:
yeah, you're right. I've changed to the new logic:
1. if has impl in cpp dispathcer, just return if the kernel is fall through kernel.
2. otherwise, if key not in py_kernels, we'll fallthrough the default ones. Or if it's py_impled with fallthrough_kernel, we can also fallthrough it.